python/ray/train/tests/test_trainer.py

```diff
@@ -1,1301 +0,0 @@
-
-    assert trainer.best_checkpoint_path is not None
-    if logdir is not None:
-        assert trainer.logdir == Path(logdir).expanduser().resolve()
-    assert trainer.latest_checkpoint_dir.is_dir()
-    assert trainer.best_checkpoint_path.is_dir()
-    assert trainer.best_checkpoint_path.name == f"checkpoint_{2:06d}"
-    assert trainer.latest_checkpoint["loss"] == 5
-    assert trainer.best_checkpoint["loss"] == 3
-
-    checkpoint_dir = trainer.latest_checkpoint_dir
-    file_names = [f.name for f in checkpoint_dir.iterdir() if f.is_dir()]
-    assert len(file_names) == 2
-    assert f"checkpoint_{2:06d}" in file_names
-    assert f"checkpoint_{3:06d}" not in file_names
-    assert f"checkpoint_{4:06d}" in file_names
-
-    def validate():
-        checkpoint = train.load_checkpoint()
-        assert checkpoint is not None
-        assert checkpoint["loss"] == 3
-
-    trainer.run(validate, checkpoint=trainer.best_checkpoint_path)
-
-
-def test_load_checkpoint_from_path(ray_start_2_cpus, tmpdir):
-    config = TestConfig()
-
-    checkpoint_strategy = CheckpointConfig(
-        checkpoint_score_attribute="loss", checkpoint_score_order="min"
-    )
-
-    def train_func_checkpoint():
-        train.save_checkpoint(loss=3)
-        train.save_checkpoint(loss=7)
-
-    trainer = Trainer(config, num_workers=2, logdir=tmpdir)
-    trainer.start()
-    trainer.run(train_func_checkpoint, checkpoint_strategy=checkpoint_strategy)
-
-    assert trainer.best_checkpoint["loss"] == 3
-    assert (
-        Trainer.load_checkpoint_from_path(trainer.best_checkpoint_path)
-        == trainer.best_checkpoint
-    )
-
-
-def test_persisted_checkpoint_strategy_failure(ray_start_2_cpus):
-    logdir = "/tmp/test/trainer/test_persisted_checkpoint_strategy_failure"
-    config = TestConfig()
-
-    def train_func():
-        train.save_checkpoint(epoch=0)
-
-    trainer = Trainer(config, num_workers=2, logdir=logdir)
-    trainer.start()
-
-    with pytest.raises(ValueError):
-        trainer.run(train_func, checkpoint_strategy=CheckpointConfig(num_to_keep=-1))
-
-    with pytest.raises(ValueError):
-        trainer.run(
-            train_func,
-            checkpoint_strategy=CheckpointConfig(
-                checkpoint_score_order="invalid_order"
-            ),
-        )
-
-    with pytest.raises(ValueError):
-        trainer.run(
-            train_func,
-            checkpoint_strategy=CheckpointConfig(
-                checkpoint_score_attribute="missing_attribute"
-            ),
-        )
-
-
-def test_world_rank(ray_start_2_cpus):
-    config = TestConfig()
-
-    def train_func():
-        return train.world_rank()
-
-    trainer = Trainer(config, num_workers=2)
-    trainer.start()
-    results = trainer.run(train_func)
-
-    assert set(results) == {0, 1}
-
-
-def test_torch_auto_unwrap(ray_start_2_cpus):
-    """Tests if underlying model from DDP is extracted when saving ckpt."""
```

Comment:
let's still test this?

Reply:
It is technically implicitly tested in e2e tests (with predictor), but I readded this.